lib/SearchProvider.js

```diff
@@ -0,0 +1,69 @@
+/* global XPCOMUtils, ContentSearch, Task, Services, EventEmitter */
+
+"use strict";
+const {Ci, Cu} = require("chrome");
+
+const CURRENT_ENGINE = "browser-search-engine-modified";
+
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.import("resource://gre/modules/Task.jsm");
+Cu.import("resource://gre/modules/Services.jsm");
+
+XPCOMUtils.defineLazyModuleGetter(this, "ContentSearch",
+                                  "resource:///modules/ContentSearch.jsm");
+
+XPCOMUtils.defineLazyGetter(this, "EventEmitter", function() {
+  const {EventEmitter} = Cu.import("resource://devtools/shared/event-emitter.js", {});
+  return EventEmitter;
+});
+
+let NewTabSearchProvider = function NewTabSearchProvider() {
+  EventEmitter.decorate(this);
+};
+
+NewTabSearchProvider.prototype = {
+
+  observe(subject, topic, data) { // jshint unused:false
```

Comment:
Also, we use ESLint (instead of JSHint), so the `jshint unused` is, in fact, unused.